### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by Donard97**
**I created this PR to check if my project is as requested.** 👀 👀

_In this project I added forms for adding posts and comments._ 💯 💯 

### Project requirements that are done:
 

> - Created form to create a post on behalf of the current_user ☑️ 
> - Created form to create a comment on behalf of the current_user ☑️ 

<hr>

_Feel free to review 👀  my project and let me know if there are any essential changes 💪_

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @Donard97, 

**This is a great project you have going on, well-done lad**. 

   
**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

## Required Changes ♻️

- [x] A user should be able to like a post. Add a likes controller file with the necessary logic. Also add the necessary create route for the likes.

_Check the inline comments under the review for sections to improve._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      format.html do
+        if @comment.save
+          redirect_to user_post_path(@post.author.id, @post.id), notice: 'Comment created successfully'
+        else
+          redirect_to user_post_path(@post.author.id, @post.id), alert: 'Comment not created, try again!'
+        end
+      end
+    end
+  end
+end
```
Comment (mke2111) [inline app/controllers/comments_controller.rb]: - [x] You are currently missing the comment#new action. Let's add this action method for new comments.
- [x] It is a good practice to keep the user input parameters in private methods. For this case, this parameter is the input text from the user. You could do something like this.
```
private 
  def comment_params
    params.require(:comment).permit(:text)
  end

```

You can then pass this as an argument to the create action
[Use this lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) to learn more about this and see how best to apply this to all your controller files.

File: config/routes.rb [hunk 2]
```diff
@@ -4,6 +4,8 @@
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
   root 'users#index'
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :comments, only: [:create]
```
Comment (mke2111) [inline config/routes.rb]: - [x] Once you add the new action method in the comments controller, allow the new route as part of this line.

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (Donard97) [reply app/controllers/comments_controller.rb]: Thank you for letting me know. I did it. 👍

File: config/routes.rb (hunk 2 above)
Comment (Donard97) [reply config/routes.rb]: Thank you for letting me know. I changed this too. 👍

**Comment by Davidosky007** [comment]
## Hi @Donard97 👋, 

## Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

done-gif


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**